src/System.ComponentModel.TypeConverter/tests/Design/DesignerVerbTests.cs

```diff
@@ -2,56 +2,153 @@
+            DictionaryEntry entry = Assert.IsType<DictionaryEntry>(Assert.Single(verb.Properties));
+            Assert.Equal("Text", entry.Key);
+            Assert.Equal(expectedPropertiesText, entry.Value);
+            Assert.True(verb.Supported);
+            Assert.True(verb.Visible);
         }
 
-        public static IEnumerable<object[]> Ctor_Text_EventHandler_CommandID_TestData()
+        public static IEnumerable<object[]> Ctor_String_EventHandler_CommandID_TestData()
         {
-            yield return new object[] { "Text", new EventHandler(EventHandler), new CommandID(Guid.NewGuid(), 10), "Text" };
-            yield return new object[] { "(&.)Text", new EventHandler(EventHandler), new CommandID(Guid.NewGuid(), 10), "Text" };
-            yield return new object[] { null, null, null, null };
+            yield return new object[] { "Text", new EventHandler(EventHandler), new CommandID(Guid.NewGuid(), 10), "Text", "Text" };
+            yield return new object[] { "(&.)Text", new EventHandler(EventHandler), new CommandID(Guid.NewGuid(), 10), "Text", "Text" };
+            yield return new object[] { null, null, null, string.Empty, null };
         }
 
         [Theory]
-        [MemberData(nameof(Ctor_Text_EventHandler_CommandID_TestData))]
-        public void Ctor_Text_EventHandler_CommandID(string text, EventHandler handler, CommandID commandID, string expectedText)
+        [MemberData(nameof(Ctor_String_EventHandler_CommandID_TestData))]
+        public void Ctor_String_EventHandler_CommandID(string text, EventHandler handler, CommandID commandID, string expectedText, string expectedPropertiesText)
         {
             var verb = new DesignerVerb(text, handler, commandID);
-            Assert.Equal(expectedText ?? string.Empty, verb.Text);
-            Assert.Same(commandID, verb.CommandID);
+            Assert.Equal(commandID, verb.CommandID);
             Assert.Empty(verb.Description);
+            Assert.True(verb.Enabled);
+            Assert.Equal(3, verb.OleStatus);
+            Assert.True(verb.Enabled);
+            Assert.Equal(expectedText, verb.Text);
+            Assert.IsType<HybridDictionary>(verb.Properties);
+            Assert.Same(verb.Properties, verb.Properties);
+            DictionaryEntry entry = Assert.IsType<DictionaryEntry>(Assert.Single(verb.Properties));
+            Assert.Equal("Text", entry.Key);
+            Assert.Equal(expectedPropertiesText, entry.Value);
+            Assert.True(verb.Supported);
+            Assert.True(verb.Visible);
+        }
+
+        [Fact]
+        public void Ctor_NullProperties_ThrowsNullReferenceException()
+        {
+            Assert.Throws<NullReferenceException>(() => new NullPropertiesDesignerVerb("Text", new EventHandler(EventHandler)));
+            Assert.Throws<NullReferenceException>(() => new NullPropertiesDesignerVerb("Text", new EventHandler(EventHandler), new CommandID(Guid.NewGuid(), 10)));
```

Comment:
Should this be fixed?

Reply:
Where is the NullReferenceException getting thrown? Looks odd that we would be expecting NRE here.